fineract-core/src/main/java/org/apache/fineract/commands/service/CommandSourceService.java

```diff
@@ -48,46 +48,34 @@ public class CommandSourceService {
     private final CommandSourceRepository commandSourceRepository;
     private final ErrorHandler errorHandler;
 
-    @Transactional(propagation = Propagation.REQUIRES_NEW, isolation = Isolation.REPEATABLE_READ)
-    public CommandSource saveInitial(CommandWrapper wrapper, JsonCommand jsonCommand, AppUser maker, String idempotencyKey) {
-        return saveInitialInternal(wrapper, jsonCommand, maker, idempotencyKey);
-    }
-
-    public CommandSource saveInitialNoTransaction(CommandWrapper wrapper, JsonCommand jsonCommand, AppUser maker, String idempotencyKey) {
```

Comment:
Can you please help me understand why we dont need anymore the "no transaction" version of this anymore?

Reply:
There is no such thing to save something without transaction, but we can reuse the existing transaction (if there is any) and the JPA can handle it, we do not need to set it by hand.
This is an ongoing discussion actually, because using the existing transaction can cause to not save the command at all, which we, I think, can not allow. We will have a discussion today. If you have time, please join.